### PR TITLE
Fix `qml.GellMann` observable data

### LINKED ISSUE #4341: [BUG] op_math adding two GellMann observables doesn't treat them as different Hermitians
### Expected behavior

```
>>> gell_1 = qml.GellMann(wires=0, index=3)
>>> gell_2 = qml.GellMann(wires=0, index=8)
>>> ham_added = gell_1 + gell_2
>>> print(type(ham_added))
<class 'pennylane.ops.qubit.hamiltonian.Hamiltonian'>
>>> print(ham_added.coeffs)
array([1,1])
>>> print(ham_added.ops)
[GellMann3(wires=[0]), GellMann8(wires=[0])]
```

The two GellMann observables should be appended in a Hamiltonian object as shown.

### Actual behavior

```
>>> gell_1 = qml.GellMann(wires=0, index=3)
>>> gell_2 = qml.GellMann(wires=0, index=8)
>>> ham_added = gell_1 + gell_2
>>> print(type(ham_added))
<class 'pennylane.ops.qubit.hamiltonian.Hamiltonian'>
>>> print(ham_added.coeffs)
array([2])
>>> print(ham_added.ops)
[GellMann3(wires=[0])]
```
Currently when adding two GellMann observables Pennylane interprets them as the same type and concatinates them together taking the first as it's type.

### Additional information

By using ```qml.operation.enable_new_opmath()``` the bahaviour seems to work correctly.

```
>>> qml.operation.enable_new_opmath()

>>> gell_1 = qml.GellMann(wires=0, index=3)
>>> gell_2 = qml.GellMann(wires=0, index=8)
>>> ham_added = gell_1 + gell_2
>>> print(type(ham_added))
<class 'pennylane.ops.op_math.sum.Sum'>
>>> print(ham_added)
GellMann3(wires=[0]) + GellMann8(wires=[0])
```

Similar to #3114, on comment chain I was told to make a new issue

### Source code

_No response_

### Tracebacks

_No response_

### System information

```shell
Name: PennyLane
Version: 0.32.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning

Platform info:           Linux-5.15.107+-x86_64-with-glibc2.31
Python version:          3.10.12
Numpy version:           1.22.4
Scipy version:           1.10.0
Installed devices:
- default.gaussian (PennyLane-0.32.0.dev0)
- default.mixed (PennyLane-0.32.0.dev0)
- default.qubit (PennyLane-0.32.0.dev0)
- default.qubit.autograd (PennyLane-0.32.0.dev0)
- default.qubit.jax (PennyLane-0.32.0.dev0)
- default.qubit.tf (PennyLane-0.32.0.dev0)
- default.qubit.torch (PennyLane-0.32.0.dev0)
- default.qutrit (PennyLane-0.32.0.dev0)
- null.qubit (PennyLane-0.32.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.31.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Fixes #4341 .

**Description of the Change:**
* Update `Observable._obs_data()` to include the index of `qml.GellMann` in the parameter list.
* Update `Hamiltonian._obs_data()` to include the index of `qml.GellMann` in the parameter list.

<ins>**Note:**</ins>
I would still recommend using the new operator arithmetic with qutrits. The generators of parametric qutrit gates are instances of `Prod` and `Sum`, not `qml.Hamiltonian`. So, try using `qml.dot` instead of `qml.Hamiltonian` with qutrit observables.

**Benefits:**
Comparison of instances of `qml.GellMann`, `Hamiltonian`s and `Tensor`s containing `qml.GellMann` is correct.

**Possible Drawbacks:**

**Related GitHub Issues:**
#4341

### CODE REVIEW

**Comment by mudit2812** [comment]
[sc-41597]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -130,6 +130,11 @@
 * `default.qutrit` now supports all qutrit operations used with `qml.adjoint`.
   [(#4348)](https://github.com/PennyLaneAI/pennylane/pull/4348)
 
+* The observable data of `qml.GellMann` now includes its index, allowing correct comparison
+  between instances of `qml.GellMann`, as well as Hamiltonians and Tensors
+  containing `qml.GellMann`.
+  [()]()
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:136]:
```suggestion
  [(#4366)](https://github.com/PennyLaneAI/pennylane/pull/4366)
```

**Comment by timmysilv** [comment]
being honest, I don't love seeing `qml.GellMan` in such base/unrelated files as `operation.py` and `hamiltonian.py`. Did you consider any other ways of getting this done? I do appreciate how concise the change is, but it seems like a circular dependency that we kinda just get away with today.

**Comment by mudit2812** [comment]
> it seems like a circular dependency that we kinda just get away with today.

I agree, and I wasn't particularly happy with the change. I tried just overloading `GellMann._obs_data()` but both `Hamiltonian` and `Observable` `_obs_data` supersede it. The alternate solution is to always use new op math, which I've recommended in the PR description as well.

If we want to correctly support it in `Hamiltonian`, adding a special case for it would be required as there are no other observables which justify having hyperparameters stored in `_obs_data`.

**Comment by timmysilv** [comment]
do we need to plan anything to have this removed in the future? I suspect it will happen naturally once we switch to arithmetic ops everywhere since that seems to be the only place this is used (Sum/Prod aren't even Observable objects!)

**Review by timmysilv — APPROVED** [review]
given the circumstances, adding 2 lines in each file ain't too bad. thanks for this!

**Review by albi3ro — APPROVED** [review]
Simple enough fix.

Would this actually work in a measurement?

**Comment by mudit2812** [comment]
> Would this actually work in a measurement?

It should. The issue was that calling `Hamiltonian.simplify` relied on `Hamiltonian.compare`, which was not taking the index of `GellMann` into account. Since this change is contained to one exception in terms of comparison for `GellMann`, measurements shouldn't be affected. Moreover, support for Hamiltonian measurements is basically non-existent in `default.qutrit`, so it doesn't really matter that much.

**Pull request merged**